vmod/vmod_debug.c

```diff
@@ -1331,3 +1331,62 @@ xyzzy_total_recall(VRT_CTX)
 
 	return (wrong);
 }
+
+/*---------------------------------------------------------------------*/
+
+struct VPFX(debug_caller) {
+       unsigned        magic;
+#define DEBUG_CALLER_MAGIC 0xb47f3449
+       VCL_SUB         sub;
+};
+
+VCL_VOID v_matchproto_(td_xyzzy_debug_caller__init)
+xyzzy_caller__init(VRT_CTX, struct VPFX(debug_caller) **callerp,
+                  const char *name, VCL_SUB sub)
+{
+       struct VPFX(debug_caller) *caller;
+
+       CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
+       AN(callerp);
+       AZ(*callerp);
+       AN(name);
+       AN(sub);
+
+       ALLOC_OBJ(caller, DEBUG_CALLER_MAGIC);
+       AN(caller);
+       *callerp = caller;
+       caller->sub = sub;
+}
+
+VCL_VOID v_matchproto_(td_xyzzy_debug_caller__fini)
+xyzzy_caller__fini(struct VPFX(debug_caller) **callerp)
+{
+       struct VPFX(debug_caller) *caller;
+
+       if (callerp == NULL || *callerp == NULL)
+               return;
+       CHECK_OBJ(*callerp, VRT_CTX_MAGIC);
```

Comment:
As mentioned under the nigoroll account, this should have been DEBUG_CALLER_MAGIC (my bad).

Reply:
interesting that it did not trip

Reply:
ah we never discard the vcl in the vtc